src/awareness.ts

```diff
@@ -92,12 +93,11 @@ export class Awareness extends ObservableV2<Events> implements YJSAwareness {
     }
     this.states = states;
 
-    if (added.length || removed.length || updated.length) {
-      // TODO(arv): YJS Awareness checks for deep equality here. We should do the
-      // same.
-      // https://github.com/yjs/y-protocols/blob/a4dd5de756f1061eaca9c1b4ebb2013e74987a11/awareness.js#L131-L135
-      this.emit('change', [{added, updated, removed}, 'local']);
-      this.emit('update', [{added, updated, removed}, 'local']);
+    if (added.length || removed.length || changed.length) {
+      this.emit("change", [{ added, updated: changed, removed }, "local"]);
+      // NOTE: with reflect we can't tell if something was set to the same value
```

Comment:
I feel like this means that we do not need the if check here since the only way we can get here is if the present clients changed.